compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmDesugarPhase.java

```diff
@@ -160,275 +154,120 @@ private static void rewriteRecordInitFunction(BIRFunction func, BRecordType reco
         func.localVars = updatedLocalVars;
     }
 
-    private static BIRFunctionParameter getFunctionParam(BIRFunctionParameter localVar) {
-        if (localVar == null) {
-            throw new BLangCompilerException("Invalid function parameter");
-        }
-
-        return localVar;
+    static void encodeModuleIdentifiers(BIRNode.BIRPackage module) {
+        encodePackageIdentifiers(module.packageID);
+        encodeGlobalVariableIdentifiers(module.globalVars);
+        encodeImportedGlobalVariableIdentifiers(module.importedGlobalVarsDummyVarDcls);
+        encodeFunctionIdentifiers(module.functions);
+        encodeTypeDefIdentifiers(module.typeDefs);
     }
 
-    private JvmDesugarPhase() {
+    private static void encodePackageIdentifiers(PackageID packageID) {
+        packageID.orgName = Names.fromString(encodeNonFunctionIdentifier(packageID.orgName.value));
+        packageID.name = Names.fromString(encodeNonFunctionIdentifier(packageID.name.value));
     }
 
-    static HashMap<String, String> encodeModuleIdentifiers(BIRNode.BIRPackage module) {
-        HashMap<String, String> encodedVsInitialIds = new HashMap<>();
-        encodePackageIdentifiers(module.packageID, encodedVsInitialIds);
-        encodeGlobalVariableIdentifiers(module.globalVars, encodedVsInitialIds);
-        encodeImportedGlobalVariableIdentifiers(module.importedGlobalVarsDummyVarDcls, encodedVsInitialIds);
-        encodeFunctionIdentifiers(module.functions, encodedVsInitialIds);
-        encodeTypeDefIdentifiers(module.typeDefs, encodedVsInitialIds);
-        return encodedVsInitialIds;
-    }
-
-    private static void encodePackageIdentifiers(PackageID packageID, HashMap<String, String> encodedVsInitialIds) {
-        packageID.orgName = Names.fromString(encodeNonFunctionIdentifier(packageID.orgName.value,
-                                                                         encodedVsInitialIds));
-        packageID.name = Names.fromString(encodeNonFunctionIdentifier(packageID.name.value, encodedVsInitialIds));
-    }
-
-    private static void encodeTypeDefIdentifiers(List<BIRTypeDefinition> typeDefs,
-                                                 HashMap<String, String> encodedVsInitialIds) {
+    private static void encodeTypeDefIdentifiers(List<BIRTypeDefinition> typeDefs) {
         for (BIRTypeDefinition typeDefinition : typeDefs) {
-            typeDefinition.type.tsymbol.name = Names.fromString(encodeNonFunctionIdentifier(
-                    typeDefinition.type.tsymbol.name.value, encodedVsInitialIds));
-            typeDefinition.internalName =
-                    Names.fromString(encodeNonFunctionIdentifier(typeDefinition.internalName.value,
-                            encodedVsInitialIds));
+            Name encodedName = Names.fromString(encodeNonFunctionIdentifier(
```

Comment:
We can also move the `Names.fromString` part inside the `encodeNonFunctionIdentifier` and `encodeFunctionIdentifier` private methods